HO: There is no significant difference of Sun Light and Water Frequency on Height of plant
H1: There is significant difference of Sun Light and Water Frequency on Height of plant

In [2]:
import numpy as np
import pandas as pd
import statsmodels.api as statm
from statsmodels.formula.api import ols

In [7]:
dt=pd.read_excel('E:/CUTM/4th sem/Marketing Analysis/Tree Height.xlsx')

In [8]:
dt

Water   Sun  Height
0    Daily   Low       6
1    Daily   Mid       6
2    Daily   Low       5
3    Daily   Mid       6
4   Weekly  High       7
5    Daily   Mid       6
6   Weekly   Low       5
7   Weekly   Mid       5
8    Daily  High       6
9   Weekly  High       7
10  Weekly  High       6
11   Daily   Mid       6
12   Daily  High       8
13  Weekly  High       8
14   Daily  High       6

In [11]:
ano=ols('Height~Water',data=dt).fit()

In [12]:
first=statm.stats.anova_lm(ano,type=2)

In [13]:
first

df     sum_sq   mean_sq         F    PR(>F)
Water      1.0   0.177778  0.177778  0.189091  0.670806
Residual  13.0  12.222222  0.940171       NaN       NaN

In [15]:
ano2=ols('Height~Sun',data=dt).fit()

In [16]:
Second=statm.stats.anova_lm(ano2,type=2)

In [17]:
Second

df   sum_sq   mean_sq        F    PR(>F)
Sun        2.0  6.07619  3.038095  5.76506  0.017593
Residual  12.0  6.32381  0.526984      NaN       NaN

Two way Anova

In [19]:
ano2=ols('Height~Sun+Water',data=dt).fit()
Two=statm.stats.anova_lm(ano2,type=2)
Two

df    sum_sq   mean_sq         F    PR(>F)
Sun        2.0  6.076190  3.038095  5.368909  0.023603
Water      1.0  0.099259  0.099259  0.175409  0.683409
Residual  11.0  6.224551  0.565868       NaN       NaN

In [20]:
an=ols('Height~Water+Sun+Water*Sun', data=dt).fit()
Int=statm.stats.anova_lm(an,type=2)
Int

df    sum_sq   mean_sq         F    PR(>F)
Water      1.0  0.177778  0.177778  0.309677  0.591444
Sun        2.0  5.997671  2.998836  5.223778  0.031203
Water:Sun  2.0  1.057884  0.528942  0.921383  0.432475
Residual   9.0  5.166667  0.574074       NaN       NaN

Water P=0.59 greater than alpha, can't reject H0
Sun, P=0.031 less than alpha reject, can reject H0
SV:FV,P=0.4324 greater than alpha can't reject H0
There is a significant difference of Sun Light on Plant height